client/asset/firo/block_deserialize.go

```diff
@@ -0,0 +1,96 @@
+package firo
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+)
+
+const (
+	// Only blocks mined with progpow are considered.
+	// Previous mining algorithms: MTP and Lyra2z ignored as being too early for
+	// Firo wallet on Dex ~ late 2023
+	ProgpowStartTime        = 1635228000 // Tue Oct 26 2021 06:00:00 UTC+0
+	HeaderLength            = 80
+	ProgpowExtraLength      = 40
+	ProgpowFullHeaderLength = HeaderLength + ProgpowExtraLength
+)
+
+// deserializeBlock deserializes the wire bytes passed in as blk and returns the
+// header for the network plus any Transparent transactions found parsed into a
+// wire.MsgBlock.
+//
+// Other transaction types are discarded; including coinbase.
+func deserializeBlock(net *chaincfg.Params, blk []byte) (*wire.MsgBlock, error) {
+	var hdrHash chainhash.Hash
+	var hdr *wire.BlockHeader
+
+	// hash header
+	var header []byte
+	switch net.Name {
+	case "mainnet", "testnet3", "testnet":
+		header = make([]byte, ProgpowFullHeaderLength)
+		copy(header, blk[:ProgpowFullHeaderLength])
+	case "regtest":
+		header = make([]byte, HeaderLength)
+		copy(header, blk[:HeaderLength])
+	default:
+		return nil, fmt.Errorf("unknown net: %s", net.Name)
+	}
```

Comment:
We should check that `len(blk) >  ProgpowFullHeaderLength ` or `len(blk) >  HeaderLength `.